### PR TITLE
Add clarification for Redux <-> Nested Navigators

### PR DESCRIPTION
**Pull request by hilkeheremans**
Insert clarification in docs on Redux and nested Navigators per https://github.com/react-community/react-navigation/issues/16

### CODE REVIEW

**Comment by satya164** [comment]
cc @mkonicek

File: docs/guides/Redux-Integration.md [hunk 1]
```diff
@@ -44,3 +44,5 @@ class App extends React.Component {
 Now, your navigation state is stored with redux, and you can fire navigation actions using redux.
 
 When a navigator is given a `navigation` prop, it relinquishes control of the state. So you are now responsible for persisting state, handling deep linking, integrating the back button, etc.
+
+If you are using nested Navigators as screens, the state of the nested navigators is also included when you attach the parent. There is no need to attach the nested Navigators to Redux directly.
```
Comment (satya164) [inline docs/guides/Redux-Integration.md]: Maybe

```
When you are nesting navigators, the navigation state is automatically passed down as long as a navigator is attached as a screen. For example:

\`\`\`js
StackNavigator({
  Home: { screen: TabNavigator },
});
\`\`\`

When you attach `StackNavigator` to Redux, `TabNavigator` automatically gets attached and you don't have to attach it separately.
```

**Comment by hilkeheremans** [comment]
Good point. Updated per your comment!

**Comment by satya164** [comment]
@mkonicek can you take a final look before merge? :D

File: docs/guides/Redux-Integration.md [hunk 2]
```diff
@@ -44,3 +44,13 @@ class App extends React.Component {
 Now, your navigation state is stored with redux, and you can fire navigation actions using redux.
 
 When a navigator is given a `navigation` prop, it relinquishes control of the state. So you are now responsible for persisting state, handling deep linking, integrating the back button, etc.
+
+When you are nesting navigators, the navigation state is automatically passed down as long as a navigator is attached as a screen. For example:
```
Comment (knowbody) [inline docs/guides/Redux-Integration.md]: It is a bit hard for me to understand when I read this sentence, I've read it a few times to get what you mean.

I would phrase it something like:

The navigation state is passed down from one navigator to another when you nest them. The child navigator, in order to receive the state from its parent, needs to be defined as a `screen`.

File: docs/guides/Redux-Integration.md [hunk 3]
```diff
@@ -44,3 +44,13 @@ class App extends React.Component {
...
+
+```js
+StackNavigator({
+  Home: { screen: TabNavigator },
+});
+```
+
+When you attach `StackNavigator` to Redux, `TabNavigator` automatically gets attached and you don't have to attach it separately.
```
Comment (knowbody) [inline docs/guides/Redux-Integration.md]: Again in here it's hard for me to understand what does it mean: to attach `StackNavigator` to Redux?

This last sentence is very unclear to me.

Comment (satya164) [inline docs/guides/Redux-Integration.md]: @knowbody maybe `connect` is a better word? I

File: docs/guides/Redux-Integration.md [hunk 4]
```diff
@@ -44,3 +44,13 @@ class App extends React.Component {
 Now, your navigation state is stored with redux, and you can fire navigation actions using redux.
 
 When a navigator is given a `navigation` prop, it relinquishes control of the state. So you are now responsible for persisting state, handling deep linking, integrating the back button, etc.
+
+When you are nesting navigators, the navigation state is automatically passed down as long as a navigator is attached as a screen. For example:
+
+```js
+StackNavigator({
+  Home: { screen: TabNavigator },
```
Comment (satya164) [inline docs/guides/Redux-Integration.md]: nit: this should be `MyTabNavigator`

**Comment by satya164** [comment]
Any updates on this?

**Comment by hilkeheremans** [comment]
Thanks for the reminder @satya164 -- I updated the docs a little per your comments + some extra.

**Comment by grabbou** [comment]
I think docs are always going to cause a massive amount of iterating as it's hard to satisfy everyone. Let's merge it and optionally wait for follow-up PRs by others that feel it can be further simplified?

**Comment by hilkeheremans** [comment]
Fine by me. I'd say we have a decent net improvement here and that is what matters most at the moment?

**Comment by grabbou** [comment]
Yeah, I'll merge it as it's 13 days old. Also, maybe even if it's unclear, it's still better than what we have now.

**Pull request merged**